components/Input/SearchMultiSelectInput/index.js

```diff
@@ -346,11 +358,14 @@ class SearchMultiSelectInput extends React.PureComponent {
 
         const finalSearchValue = searchValue || '';
 
-        const filteredOptions = this.filterOptions(
+        const filteredOptions = this.getFilteredOptions(
             options,
             labelSelector,
+            keySelector,
             searchValue,
+            value,
```

Comment:
Sending value to the filteredOptions causes a new list to be created whenever a value is selected which results in scroll jumps and unnecessary list calculations.